code/game/objects/items/weapons/material/knives.dm

```diff
@@ -110,3 +110,47 @@
 	name = "lightweight utility knife"
 	desc = "A lightweight utility knife made out of a steel alloy."
 	icon_state = "titanium"
+
+//Reagent Knives
+
+/obj/item/material/knife/reagent
+	atom_flags = ATOM_FLAG_OPEN_CONTAINER
+	origin_tech = list(TECH_MATERIAL = 2, TECH_ESOTERIC = 5)
+
+/obj/item/material/knife/reagent/New()
+	..()
+	create_reagents(30)
+
+/obj/item/material/knife/reagent/attack(mob/living/M, mob/user, var/target_zone)
+	if(!istype(M))
+		return
+
+	. = ..()
+
+	var/allow = M.can_inject(user, target_zone)
+	if(user.a_intent == I_HURT) //prevents it from injecting reagents WITHOUT attacking (reagents get injected but no harm is done to the target)
+		if(allow)
+			if(allow == INJECTION_PORT) //this is a knife meant for combat, we are not gonna inject
+				return
+		if(reagents.total_volume)
+			if(M.reagents)
+				var/should_admin_log = reagents.should_admin_log()
+				var/contained_reagents = reagents.get_reagents()
+				var/trans = reagents.trans_to_mob(M, 4, CHEM_BLOOD) //we transfer only 4 units because slashing doesnt generally make you stay in the wound long
```

Comment:
More of a nitpick, but you might want to make starting max volume, and amount of chems transfered a variable.
Also, transfering 4u of ANY possible reagent per hit on a somewhat fast weapon is rather strong, might want to reconsider it.

By the by, did you check if it can be refilled?

Reply:
1 unit?

Reply:
also, yes, it can be refilled